### PR TITLE
GOONCHEM

### PR DESCRIPTION
**Pull request by Iamgoofball**
Adds Goon Chemistry to the game. 50+ new chemicals, too many to list, a chemical heater system,  CLF3, and chemical patches.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
So it begins.

**Comment by Iamgoofball** [comment]
muh tricord!!!

**Comment by fj45** [comment]
Welp, there it is. 


I shall be in my bunker.

**Comment by Artorp** [comment]
Uh oh.

**Comment by Flip-Sandwich** [comment]
Hi, I'm a Goon Chemist and this makes me giddy, because I can't wait until I can use [gibsmoke].

**Comment by tigercat2000** [comment]
Interesting, but uh, why remove all the FILE_DIR definitions? That seems like it would definitely break something.

**Comment by ZomgPonies** [comment]
The FILE_DIR shit is because most of you have your Dreamdaemons configured improperly

Build -> Preferences -> Automatically set FILE_DIR for sub-directories allows you to use relative paths, which is a terrible, terrible thing. Always use absolute paths in your code.

**Comment by ZomgPonies** [comment]
Closing this PR because it's WGW in russian. If you can't be arsed to follow our contributor guidelines, we can't be arsed to merge your shit in.

**Comment by Iamgoofball** [comment]
@ZomgPonies Yeah, wgw was a place holder, I threw this PR up at 3 in the morning and was about to pass out. I'll make a nicer description.

**Comment by TheDZD** [comment]
I'm assuming this will be replacing Star Trek chems at some point, rather than just being an addition?

**Comment by Iamgoofball** [comment]
@TheDZD yep

see: branch name

File: code/modules/reagents/newchem/medicine.dm [hunk 1]
```diff
@@ -0,0 +1,837 @@
...
+		if(prob(30))
+			M.eye_blind = 0
+		else if(prob(80))
+			M.eye_blind = 0
+			M.eye_blurry = 1
+		if(M.eye_blurry > 0)
+			if(prob(80))
+				M.eye_blurry = 0
+	..()
+	return
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/medicine.dm]: Due to how we have internal organs and jazz, I recommend using this for oculine instead: https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/reagents/Chemistry-Reagents.dm#L1758

File: code/modules/reagents/newchem/toxins.dm [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+	id = "pancuronium"
+	description = "Knocks you out after 30 seconds, 7% chance to cause some oxygen loss."
+	reagent_state = LIQUID
+	color = "#CF3600"
+	metabolization_rate = 0.2
+
+datum/reagent/pancuronium/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(current_cycle >= 10)
+		M.SetParalysis(3)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/toxins.dm]: weaken instead of paralysis.

File: code/modules/reagents/newchem/toxins.dm [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+	id = "neurotoxin2"
+	description = "Deals toxin and brain damage up to 60 before it slows down, causing confusion and a knockout after 17 elapsed cycles."
+	reagent_state = LIQUID
+	color = "#CF3600"
+	var/cycle_count = 0
+	metabolization_rate = 1
+
+datum/reagent/neurotoxin2/on_mob_life(var/mob/living/M as mob)
+	cycle_count++
+	if(M.brainloss + M.toxloss <= 60)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/toxins.dm]: TOX loss occurs every cycle; brain damage is the only one that stops at 60.

File: code/modules/reagents/newchem/medicine.dm [hunk 4]
```diff
@@ -0,0 +1,837 @@
...
+	id = "pen_acid"
+	description = "Reduces massive amounts of radiation and toxin damage while purging other chemicals from the body. Has a chance of dealing brute damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+
+datum/reagent/pen_acid/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.radiation > 0)
+		M.radiation -= 7
+	M.adjustToxLoss(-4*REM)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/medicine.dm]: make this a 70% chance to reduce tox loss by 4; closest approximation to 70% of cycles, guaranteed.

File: code/modules/reagents/newchem/medicine.dm [hunk 5]
```diff
@@ -0,0 +1,837 @@
...
+	description = "Has a 33% chance per metabolism cycle to heal brute and burn damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+
+datum/reagent/salglu_solution/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(prob(33))
+		M.adjustBruteLoss(-1*REM)
+		M.adjustFireLoss(-1*REM)
+	..()
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/medicine.dm]: These are both 2 now.

**Comment by Fox-McCloud** [comment]
Overall, looks good; aside from a few small changes, it looks like this is pretty much ready to go.

File: code/modules/reagents/newchem/chem_heater.dm [hunk 6]
```diff
@@ -0,0 +1,118 @@
+/obj/machinery/chem_heater
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/chem_heater.dm]: might as well add the board+all that jazz for this--unless that's for a future PR.

**Comment by ZomgPonies** [comment]
Once Fox gives the green light on code quality, this can be merged right away. It's already been discussed ad nauseum on the forums

File: code/modules/reagents/newchem/chem_heater.dm (hunk 6 above)
Comment (Iamgoofball) [inline code/modules/reagents/newchem/chem_heater.dm]: future PR, every single codebase has special snowflake construction and effects, and I'll do a deeper analysis into how yours works in the future for this.

**Comment by ZomgPonies** [comment]
Merge conflict.

File: code/modules/mob/living/carbon/human/life.dm [hunk 7]
```diff
@@ -812,7 +814,7 @@ var/global/list/brutefireloss_overlays = list("1" = image("icon" = 'icons/mob/sc
 
 	proc/handle_chemicals_in_body()
 
-		if(reagents && !(species.flags & IS_SYNTHETIC)) //Synths don't process reagents.
+		if(reagents) //Synths don't process reagents. // fuck you they do now - Iamgoofball
```
Comment (IAmBigCoat) [inline code/modules/mob/living/carbon/human/life.dm]: This is the most important change.

Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/life.dm]: everyone gets to be on an even playing field when it comes to chemicals

none of your bullshit snowflake in my chemistry code

**Comment by ZomgPonies** [comment]
Yes and no. The goal is to seperate all the reagents between stuff that affects organics, stuff that affects inorganics and stuff that affects slimes (because waterblood and what not). We'll need to assign bitflags for those and go through the list of reagents.

Also travis failed, fix your compile errors.

File: code/game/objects/items/devices/scanners.dm [hunk 8]
```diff
@@ -172,12 +189,8 @@ REAGENT SCANNER
 		OX = fake_oxy > 50 ? 		"\red Severe oxygen deprivation detected\blue" 	: 	"Subject bloodstream oxygen level normal"
 	user.show_message("[OX] | [TX] | [BU] | [BR]")
 	if (istype(M, /mob/living/carbon))
-		if(M:reagents.total_volume > 0 && upgraded)
-			user.show_message("\blue Reagents detected in subject's blood:")
-			for(var/A in M.reagents.reagent_list)
-				var/datum/reagent/R = A
-				user.show_message("\t \blue [M.reagents.get_reagent_amount(R.id)]u [R.name]")
-		if(M:virus2.len)
+		chemscan(user, M)
```
Comment (Fox-McCloud) [inline code/game/objects/items/devices/scanners.dm]: Only allow the chemscan if the analyzer is upgraded.

Comment (Iamgoofball) [inline code/game/objects/items/devices/scanners.dm]: will add a check

**Comment by ZomgPonies** [comment]
Fox, Goof says you also need to map in the heaters? Might want to do that now. Merging.

**Comment by Fox-McCloud** [comment]
I'll get the mapping

**Comment by Iamgoofball** [comment]
@Fox-McCloud get on skype pls

**Comment by fj45** [comment]
Thee end is nigh! 


Can't wait for the complete chaos this makes. Thanks for all the hard work coders!

**Comment by Fishocalypse** [comment]
I can't wait until medbay crashes and burns.


A good suggestion would be to test run the new chem system on extended.

**Pull request merged**